stdlib/mailbox.pyi

```diff
@@ -32,7 +33,9 @@ __all__ = [
 
 _T = TypeVar("_T")
 _MessageT = TypeVar("_MessageT", bound=Message)
-_MessageData: TypeAlias = email.message.Message | bytes | str | IO[str] | IO[bytes]
+_MessageData: TypeAlias = email.message.Message | bytes | str | io.StringIO | _SupportsReadAndReadline
+
+class _SupportsReadAndReadline(SupportsRead[bytes], SupportsNoArgReadline[bytes], Protocol): ...
```

Comment:
Nit: It's slightly confusing to have the alias defined before the protocol, since the alias depends on the protocol

```suggestion

class _SupportsReadAndReadline(SupportsRead[bytes], SupportsNoArgReadline[bytes], Protocol): ...

_MessageData: TypeAlias = email.message.Message | bytes | str | io.StringIO | _SupportsReadAndReadline
```

Reply:
Thanks!